crates/re_format/src/lib.rs

```diff
@@ -33,43 +64,126 @@ pub fn format_number(number: usize) -> String {
+            // perfect integer
+            format_i64(value.round() as i64)
+        } else {
+            let num_decimals = num_decimals.unwrap_or_else(|| {
+                let magnitude = value.abs().log10();
+                (3.5 - magnitude).round().max(1.0) as usize
+            });
+            let mut formatted = format!("{value:.num_decimals$}");
+
+            if strip_trailing_zeros {
+                while formatted.ends_with('0') {
+                    formatted.pop();
+                }
+                if formatted.ends_with('.') {
+                    formatted.pop();
+                }
+            }
+
+            if let Some(dot) = formatted.find('.') {
+                let integer_part = &formatted[..dot];
+                let fractional_part = &formatted[dot + 1..];
+                let integer_part = add_thousands_separators(integer_part);
+                // For the fractional part we should start counting thousand separators from the _front_, so we reverse:
+                let fractional_part = reverse(&add_thousands_separators(&reverse(fractional_part)));
+                format!("{integer_part}.{fractional_part}")
+            } else {
+                add_thousands_separators(&formatted) // it's an integer
+            }
+        }
+    }
 }
 
 /// Format a number with a decent number of decimals.
+///
+/// The returned value is for human eyes only, and can not be parsed
+/// by the normal `f64::from_str` function.
 pub fn format_f64(value: f64) -> String {
-    let is_integer = value.round() == value;
-    if is_integer {
-        return format!("{value:.0}");
-    }
+    FloatFormatOptions::default().format_f64(value)
+}
 
-    let magnitude = value.abs().log10();
-    let num_decimals = (3.5 - magnitude).round().max(1.0) as usize;
-    format!("{value:.num_decimals$}")
+fn reverse(s: &str) -> String {
+    s.chars().rev().collect()
 }
 
 /// Format a number with a decent number of decimals.
+///
+/// The returned value is for human eyes only, and can not be parsed
+/// by the normal `f64::from_str` function.
 pub fn format_f32(value: f32) -> String {
     format_f64(value as f64)
 }
 
 #[test]
 fn test_format_float() {
+    assert_eq!(format_f64(f64::NAN), "NaN");
+    assert_eq!(format_f64(f64::INFINITY), "∞");
+    assert_eq!(format_f64(f64::NEG_INFINITY), "−∞");
+    assert_eq!(format_f64(0.0), "0");
     assert_eq!(format_f64(42.0), "42");
-    assert_eq!(format_f64(123_456_789.0), "123456789");
-    assert_eq!(format_f64(123_456_789.123_45), "123456789.1");
-    assert_eq!(format_f64(0.0000123456789), "0.00001235");
-    assert_eq!(format_f64(0.123456789), "0.1235");
+    assert_eq!(format_f64(-42.0), "−42");
+    assert_eq!(format_f64(-4.20), "−4.2");
+    assert_eq!(format_f64(123_456_789.0), "123 456 789");
+    assert_eq!(format_f64(123_456_789.123_45), "123 456 789.1");
+    assert_eq!(format_f64(0.0000123456789), "0.000 012 35");
+    assert_eq!(format_f64(0.123456789), "0.123 5");
```

Comment:
Nit-picky but this value seems like a bad default to me.  I would expect this to match the behavior of the following line and only show 3 decimal places.